summingbird-online/src/main/scala/com/twitter/summingbird/online/executor/AsyncBase.scala

```diff
@@ -107,13 +136,15 @@ abstract class AsyncBase[I, O, S, D, RC](maxWaitingFutures: MaxWaitingFutures, m
     if (pendingFuturesCount > maxWaitingFuturesCount) {
       // Too many futures waiting, let's clear.
       outstandingFutures.dequeueAll(_.isDefined)
```

Comment:
maybe rather than `dequeueAll` we want to do:
```scala
val (finished, pending) = outstandingFutures.toSeq.partition(_.isDefined)
val toClear = pending.size - maxWaitingFuturesCount
if (toClear > 0) {
  // wait on N, put the ones still not defined at the end.
}
else {
  outstandingFutures.putAll(pending)
}
```
since `dequeueAll` internally does `toSeq` the above should be faster (and maybe a bit clearer).